compiler/rustc_target/src/spec/armv8r_none_eabihf.rs

```diff
@@ -0,0 +1,32 @@
+// Targets the Little-endian Cortex-R52 processor (ARMv8-R)
+
+use crate::spec::{Cc, LinkerFlavor, Lld, PanicStrategy, RelocModel, Target, TargetOptions};
+
+pub fn target() -> Target {
+    Target {
+        llvm_target: "armv8r-none-eabihf".into(),
+        pointer_width: 32,
+        data_layout: "e-m:e-p:32:32-Fi8-i64:64-v128:64:128-a:0:32-n32-S64".into(),
+        arch: "arm".into(),
+
+        options: TargetOptions {
+            abi: "eabihf".into(),
+            linker_flavor: LinkerFlavor::Gnu(Cc::No, Lld::Yes),
+            linker: Some("rust-lld".into()),
+            relocation_model: RelocModel::Static,
+            panic_strategy: PanicStrategy::Abort,
+            // The Cortex-R52 supports either fp-armv8 single-precision only with 16
+            // double-precision registers, or neon-fp-armv8 with 32 double-precision registers.
+            // Use the lesser of these two options as the default.
```

Comment:
The wording in the referenced documentation suggests that this is a "NEON is optional” rather than “you get fp XOR neon” choice. This is quite standard in other ARM targets, unlike the XOR choice.

Reply:
I'm not sure I follow, or if my comment wasn't clear. The critical part I was trying to capture here is that the non-neon version also has only 16 DP registers; there is no version that has 32 DP regs but no neon.

Reply:
Also there is no `nofp` version of the R52, as I understand the documentation.

Reply:
Ah, so my interpretation of that comment was as such:

> supports either (fp-armv8 (which is single-precision only with 16 dp regs)) or (neon-fp-armv8 (which has do 32 double precision registers)).

hence the fp-armv8 XOR neon-fp-armv8 interpretation. Would it be better to copy over the note from the [upstream documentation](https://developer.arm.com/documentation/100026/0102/Advanced-SIMD-and-floating-point-support/About-the-Advanced-SIMD-and-floating-point-support) and then amend it with information about registers if you must? For example as such:

> The Cortex-R52 processor supports single-precision floating-point instructions (fp-armv8) and can optionally support the double-precision floating-point and Advanced SIMD instructions (neon-fp-armv8).
>
> In fp-armv8 only 16 double-precision registers are available. Available set of registers expand to 32 double-precision registers when neon-fp-armv8 is enabled.

or something.

Reply:
Your interpretation is correct. I'll try and re-word it so it's clearer. The upstream web docs are IMO unclear because it's ambiguous whether neon and DP are coupled, but the manual is clear that there are exactly two configurations.

Reply:
Updated the comment describing the FP variants. Let me know if it's acceptable.

Reply:
Yeah, the new comment is quite understandable.